### PR TITLE
[7-Zip] update to 22.01

### LINKED ISSUE #27993: [7-Zip] update to 22.01
Library name: 7zip

New version number: 22.01

https://www.7-zip.org/download.html

https://sourceforge.net/p/sevenzip/discussion/45797/thread/c43cbc5f18/

This issue is only port update request.

### PR DESCRIPTION
**Pull request by JonLiu1993**
Fix #27993 

Update 7zip to the latest version 22.01

Note: no feature need to test.

### CODE REVIEW

File: ports/7zip/vcpkg.json [hunk 1]
```diff
@@ -1,7 +1,6 @@
 {
   "name": "7zip",
-  "version": "22.0",
-  "port-version": 1,
+  "version-string": "22.01",
```
Comment (JonLiu1993) [inline ports/7zip/vcpkg.json:3]: When I use the original "version", powershell tells me
````
error: while loading F:\Feature-test\v8\vcpkg\ports\7zip\vcpkg.json:
$ (a manifest): 'version' text was not a relaxed version:
error: `22.01` is not a valid relaxed version (semver with arbitrary numeric element count).
````
When I try to use "version-semver", I get the error again:
````
error: while loading F:\Feature-test\v8\vcpkg\ports\7zip\vcpkg.json:
$ (a manifest): 'version-semver' text was not a semantic version:
error: `22.01` is not a valid semantic version, consult
````
So in the end I chose "version-string"

File: ports/7zip/vcpkg.json [hunk 2]
```diff
@@ -1,7 +1,6 @@
...
   "description": "Library for archiving file with a high compression ratio.",
   "homepage": "https://www.7-zip.org",
   "license": "LGPL-2.1-or-later",
```
Comment (MonicaLiu0311) [inline ports/7zip/vcpkg.json:6]:
```suggestion
  "license": "LGPL-2.1-or-later",
```

Comment (MonicaLiu0311) [reply ports/7zip/vcpkg.json:6]: Please check the "[license](https://www.7-zip.org/license.txt)".

**Review by MonicaLiu0311 — APPROVED** [review]

File: ports/7zip/vcpkg.json (hunk 1 above)
Comment (dg0yt) [reply ports/7zip/vcpkg.json:3]: Where is the problem with `"versions": "22.1"`?
vcpkg's unwillingness to handle leading zeros causes similar fuzziness for `version-date` in port poppler et al. Today, it also limits the usability of `${VERSION}`.

**Review by dg0yt — CHANGES_REQUESTED** [review]

**Comment by JavierMatosD** [comment]
Thank you!

**Comment by dg0yt** [comment]
IMO this PR needlessly degraded the version scheme.

**Pull request merged**